app/assets/javascripts/discourse/app/instance-initializers/video-placeholder.js

```diff
@@ -0,0 +1,92 @@
+import { withPluginApi } from "discourse/lib/plugin-api";
+import { iconHTML } from "discourse-common/lib/icon-library";
+import discourseLater from "discourse-common/lib/later";
+import I18n from "I18n";
+
+export default {
+  initialize(owner) {
+    withPluginApi("0.8.7", (api) => {
+      function handleVideoPlaceholderClick(helper, event) {
+        const parentDiv = event.target.closest(".video-placeholder-container");
+        const wrapper = event.target.closest(".video-placeholder-wrapper");
+
+        const videoHTML = `
+        <video width="100%" height="100%" preload="metadata" controls style="display:none">
+          <source src="${parentDiv.dataset.videoSrc}" ${parentDiv.dataset.origSrc}>
+          <a href="${parentDiv.dataset.videoSrc}">${parentDiv.dataset.videoSrc}</a>
+        </video>`;
+        parentDiv.insertAdjacentHTML("beforeend", videoHTML);
+        parentDiv.classList.add("video-container");
+
+        const video = parentDiv.querySelector("video");
+
+        const caps = owner.lookup("service:capabilities");
+        if (caps.isSafari || caps.isIOS) {
+          const source = video.querySelector("source");
+          if (source) {
+            // In post-cooked.js, we create the video element in a detached DOM
+            // then adopt it into to the real DOM.
+            // This confuses safari, and preloading/autoplay do not happen.
+
+            // Calling `.load()` tricks Safari into loading the video element correctly
+            source.parentElement.load();
+          }
+        }
```

Comment:
Do you know if this is still necessary for Safari in iOS? What's the behaviour without it?

Reply:
I tried it on desktop safari, without it the video doesn't play /shrug

Reply:
This is what happens in safari without the source loading function.

![CleanShot 2023-10-12 at 10 19 34](https://github.com/discourse/discourse/assets/1490496/33b42f9f-26a8-4b39-af03-1d748a1a0e3c)

On click, video gets downloaded in the background, but never plays. The placeholder never goes away, because it's waiting for the `onCanPlay` event to fire. Firefox and Chrome don't need this.

Reply:
Interesting... ok!